scmrepo/git/backend/base.py

```diff
@@ -251,11 +247,6 @@ def fetch_refspecs(
             refspecs: Iterable containing refspecs to fetch.
                 Note that this will not match subkeys.
             force: If True, local refs will be overwritten.
-            on_diverged: Callback function which will be called if local ref
-                and remote have diverged and force is False. If the callback
-                returns True the local ref will be overwritten.
-                Callback will be of the form:
-                    on_diverged(local_refname, remote_sha)
```

Comment:
What's the reasoning for removing `on_diverged` support?

Reply:
In previous, `on_diverged` was only used for raising Exceptions when diverged. Because for now, we will not stop and raise this exception even if they diverged from remote. And for the force push, we already have a `force` flag for this.

Reply:
Do you have any further usage for this?

Reply:
Noticed that there was some related code in `executor` do you have any thoughts on this?

Reply:
It's a useful callback to support because the desired behavior may be to keep the existing one, without considering it a failing state. Passing the refs through the callback allows the caller to do an actual diff on the diverged commits, and decide whether or not they should force push/pull, do nothing, or do a merge/rebase as needed.

We don't use it right now in DVC, but I think we should keep support for the callback in scmrepo

Reply:
Basically with scmrepo, we should provide useful general API wrappers around the git backends, even if we don't use them in DVC